Nodejs/Product/Nodejs/Telemetry/NodejsTelemetryExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+
+using System;
+
+namespace Microsoft.NodejsTools.Telemetry {
+    /// <summary>
+    /// Extensions for logging telemetry events.
+    /// </summary>
+    internal static class NodejsTelemetryExtensions {
+        public static void LogIntelliSenseActivatedForProject(this ITelemetryLogger logger, Guid projectGuid, Options.AnalysisLevel newAnalysisLevel) {
+            logger.ReportEvent(
+                TelemetryEvents.IntelliSenseActivatedForProject,
+                new DataPointCollection(
+                    new DataPoint(TelemetryProperties.ProjectGuid, projectGuid.ToString("B")),
+                    new DataPoint(TelemetryProperties.IntelliSenseLevel, newAnalysisLevel.ToString())));
+        }
```

Comment:
I agree that we should not see any different per-project at a specific point. What this point allows us to do is track intellisense changes by project or project session for correlation, and also more easily write queries that do something like: for 90% projects, users have ES6 enabled.

Logging per project allows all the same queries as if we were to log intellisense level per VS session, plus other interesting queries using project correlation.